### PR TITLE
UCT: Add 'component' param to internal md_open() and query_resources()

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Allows reusing same functions (md_open, query_resources) for similar components

### CODE REVIEW

File: src/uct/base/uct_component.h [hunk 1]
```diff
@@ -66,7 +69,7 @@ struct uct_md_component {
                                 _cfg_prefix, _cfg_table, _cfg_struct, _cm_open) \
     \
     uct_md_component_t _mdc = { \
-        .query_resources = _query, \
+        .query_md_resources = _query, \
```
Comment (yosefe) [inline src/uct/base/uct_component.h]: didn't fix indentation here because i'm going to remove this macro in next PR

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11524/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 01-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/bc951d207db07cd4fa3e24980b5f94bf84bb0001/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/a52b05ec9a215f85d35f50e3c78a1b36ec76447e/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/7172a1a3c08e15b5168143035fbae82b70898a8e/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/01d287e3c1b90d4131997301047b293f67000dd8/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/85d52ce2ca01d7a83586bab4e7506728a7c186f8/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/0b26a8a37937780ad23fb36350ca243bc4ba4f73/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/047774d9bf45a6532432c8af309f7d707356ea4a/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/86cac0c589aa45c0887c6eb5c95db508e688c22b/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/4dbfdddb1e9e4a3c828b76c5e81424f29ae6b72e/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/ec0e9b0239a69a9148b851723b06688d3270173c/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/f8dd427e009025fa988ee2a377973253266c5d16/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/eb09804ebcf0a4057fe09d118cc6a75ffce428e0/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/c8aace6b2cff8d8e6d3baee4f6289950d61349b5/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/5ef79f8da1739d18acab0ec264f88c1f9574c6d5/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/acd36dc957da0d915921254a55800b6cdf9a83ed/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/a9b1e1c588c852827e320db239d93012cba7cf36/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/b1a6501c935c3171d235e92038eebb1720c4bc70/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/76ad627e78f39e805d83d8a29714ac2bcf1de166/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/700e0df7ff925da3b1819924091892abf987acce/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/8aa6bf31b7ceec85f90c71474e2ebc0ec150895f/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/0ba61f3dc0cea8de998e8d77efced581ac51a570/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/559489305b034940327ddc49ee670e7ade6c160c/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/95f273b81b38c0c7057ba5ba46155ae1db0e62c1/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/0eac7ee6052cd225efd38d6976abd6ad07b61285/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/daab7e2cced08c9a59eb985012f41368b0693b34/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/e0d7b1b0783e6815bbee7852fb6dba9e8591de69/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/a62da26e0d972a0840ac18337a8dff7d669b3113/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/74662459fc5fe5bb98d6b6045a90cc6b17a5d634/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/37641bad5c4d08987e83454964cb1adb/raw/14bb5bbe3eefc51c228e7c407f610068c0792388/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11545/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 03-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/570db3a7ce06c23f28c1fd3913e23faf0a59312d/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/78c255f0de6aba1ac6b9e2e3f27bfff3896e974e/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/765f5fc3ae4a4dc1bd8c5898001094133fe92ff4/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/5809e69b85d079411700259ef8a89b577961774d/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/57a86e513d741d69f0ab5aba2c666dcf23759420/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/6940649b9b3b064a3d22f792ccf7b1374cb28ccf/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/78195897234264f0bdd6bc93913afac8a4c1366e/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/9e22185f74d7f7210711a0bdbb4b6c166b9bc07b/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/472fb85b963cb8fcf10dd4e76f0fa9b3bf62b8e7/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/357ba7ae72f54c8d8cdc1c9026bb40149288ec00/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/56886d9f87dc5e5a2c5042b7258333d6b29ead30/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/f8a89bd90bdb9f95fe8ba06cecd19bab516095b2/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/acab19b040efd95383487919b6612a428c2d6a66/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/176e78b404c7a95f85a786c6782e3f6c058c9270/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/fb4c092941e6628030e3181669bafa35a1cdec7a/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/0fe270c2dd68b87820b1a51f7c75c8083b8875dc/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/d54a281212d03a107729f747067a143d20f2eabb/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/43902c7d3376654916d9a8c1355e39701abc4bf3/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/026159255fea7929fb46a9d4b1d258ee00eaf0e3/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/9fd04e8931a138a53e661fbc6c07467b160c348d/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/16dbd41fe937bb6ba941e133435731616fca5d7f/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/9b5608806a563ce891ddca5df9e701d67d8ea0ba/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/abaa660991b5a43f6a9f74bb6859669ad20614c7/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/dd0b5ef1b78a766b30133c68b84ac0ea4296f611/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/8d38449c804c7830aa51af4690376b7177e8c727/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/cdee3d323db42d555c993d7b9ba78b2a990224f8/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/9eee2ea104017ae547433bbe312c5b9f7801743c/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/e16be4328b58abadf6d75ac0db4cbd186571b552/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/8d11771ebd42cd111c4ff70fc3d45585/raw/b1456aea9b2914ca949a34acf2309b642745479f/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
bot:pipe:retest

**Pull request merged**